great_expectations/zep/context.py

```diff
@@ -104,7 +110,15 @@ def factories(self) -> List[str]:
 
     def __getattr__(self, name):
         try:
-            return self.__source_factories[name]
+            fn = self.__source_factories[name]
+
+            def wrapped(*args, **kwargs):
+                datasource = fn(*args, **kwargs)
+                # TODO (bdirks): add_zep_datasource to the AbstractDataContext class
+                self._data_context.add_zep_datasource(datasource)
+                return datasource
+
+            return wrapped
```

Comment:
## non-blocking

At first glance, I misunderstood this.
All these `add_X` methods are confusing me.

Is the another name we could use to clarify that we are attaching the newly created data-source instance to the DataContext??

Also I think it's fine to merge as and figure this out later.

Reply:
I've changed this to `_attach_datasource_to_context`.